org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/ConnectionStreamFactory.java

```diff
@@ -125,41 +155,49 @@ public OutputStream getOutputStream() throws IOException {
 
 	}
 
-	private static String OS = System.getProperty("os.name").toLowerCase();
 	private StreamProvider provider;
 
 	/**
 	 *
 	 * @return
 	 */
-	public StreamProvider getSelectedStream(){
+	public StreamProvider getSelectedStream() {
 		if (provider == null) {
```

Comment:
shouldn't we return the provider once it's instantiated? If you set both CLIENT_PORT and STDIN_PORT/STDOUT_PORT, you'll get "STDIN_PORT and STDOUT_PORT will be removed in the next release. Please use CLIENT_PORT instead (single connection for both in and output)" anyway